spring-cloud-data-rest/src/main/java/org/springframework/cloud/data/rest/controller/StreamController.java

```diff
@@ -118,10 +125,88 @@ public void save(@RequestParam("name") String name,
 			@RequestParam(value = "deploy", defaultValue = "true")
 			boolean deploy) {
 
-		StreamDefinition definition = new StreamDefinition(name, dsl);
-		this.repository.save(definition);
+		StreamDefinition stream = new StreamDefinition(name, dsl);
+		stream = this.repository.save(stream);
+		if (deploy) {
+			deployStream(stream);
+		}
+	}
+
+	/**
+	 * Request removal of an existing stream definition.
+	 *
+	 * @param name the name of an existing stream definition (required)
+	 */
+	@RequestMapping(value = "/definitions/{name}", method = RequestMethod.DELETE)
+	@ResponseStatus(HttpStatus.OK)
+	public void delete(@PathVariable("name") String name) throws Exception {
+		this.repository.delete(name);
+	}
+
+	/**
+	 * Request removal of all stream definitions.
+	 */
+	@RequestMapping(value = "/definitions", method = RequestMethod.DELETE)
+	@ResponseStatus(HttpStatus.OK)
+	public void deleteAll() throws Exception {
+		this.repository.deleteAll();
```

Comment:
both `destroy` and `destroyAll` should indeed invoke the `undeploy` commands first (at the moment, with the LocalModuleDeployer that will fail, and cause the Admin to crash - we really need to handle Exceptions within the REST API so AdminApplication does not crash!)